File: euler/1-multiples-of-3-and-5/dart-2.18/solution1.dart
int sum(int r, int n) => (n * (r + n * r)) >> 1;

// Doesn't support integers larger than 64 bits.
int multiplesOf3And5Below(int n) {
  final int e = n - 1;

  final int setA = sum(3, e ~/ 3);
  final int setB = sum(5, e ~/ 5);
  final int setC = sum(15, e ~/ 15);

  return setA + setB - setC;
}

File: euler/1-multiples-of-3-and-5/go-1.19/solution1.go
package main

type s1 struct{}

func sum(r int64, n int64) int64 {
	return (n * (r + n*r)) >> 1
}

// Fast multiples of 3 and 5 using AP [complexity O(1)].
func (s1) MultiplesOf3And5Below(n int64) int64 {
	var e int64 = n - 1

	var setA int64 = sum(3, e/3)
	var setB int64 = sum(5, e/5)
	var setC int64 = sum(15, e/15)

	return setA + setB - setC
}
